C-LinkedLists/src/SLL012Sort.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+OVERVIEW:  Given a single linked list of 0s 1s and 2s ,Sort the SLL such that zeroes
+will be followed by ones and then twos.
+
+You should only change the links and shouldnt swap the data. Basically no write operation should be done on data field.
+
+INPUTS:  SLL head pointer
+OUTPUT: Sorted SLL ,Head should Finally point to an sll of sorted 0,1,2
+ERROR CASES:
+NOTES: Only 0,1,2, will be in sll nodes
+*/
+
+#include <stdio.h>
+#include <malloc.h>
+
+struct node {
+	int data;
+	struct node *next;
+};
+
+/*
+Optional : For this function swapping data is allowed.Although this wont be tested on our end.
+It is only for learning purpose.
+*/
+void sll_012_sort_swapData(struct node **head){
+
+}
+
+/*
+This is the actual function. You are supposed to change only the links.
+*/
+void setNodes(struct node **head,struct node* head0, struct node *head1, struct node *head2, struct node *head0end, struct node *head1end, struct node *head2end) {
```

Comment:
Try to extract common parts to other smaller functions. Which will make it easier to understand when there are a couple of if-else cases. This is good but having seperate functions would be more readable and nice to look at.